test/format-test.cc

```diff
@@ -2473,3 +2473,37 @@ TEST(FormatTest, FormatUTF8Precision) {
   EXPECT_EQ(result.size(), 5);
   EXPECT_EQ(from_u8str(result), from_u8str(str.substr(0, 5)));
 }
+
+#ifdef __cpp_decltype_auto
```

Comment:
Can we make the test work without this, e.g. checking if the types are the same manually?

Reply:
Done. Just have a `static_assert`.